components/Collections/src/MDCCollectionViewController.m

```diff
@@ -213,8 +213,20 @@ - (CGSize)sizeWithAttribute:(UICollectionViewLayoutAttributes *)attr
 
 - (CGFloat)cellWidthAtSectionIndex:(NSInteger)section
                     collectionView:(UICollectionView *)collectionView {
-  CGFloat bounds =
-      CGRectGetWidth(UIEdgeInsetsInsetRect(collectionView.bounds, collectionView.contentInset));
+  UIEdgeInsets contentInset = collectionView.contentInset;
+  // On the iPhone X, we need to offset
+  if ([self.view respondsToSelector:@selector(adjustedContentInset)]) {
```

Comment:
This should be
```objectivec
if ([collectionView respondsToSelector:@selector(adjustedContentInset)]) {
```

However, I think it may be cleaner to just wrap the whole thing in a macro (for SDK checking) and availability check (for iOS checking).  What do you think?

Reply:
Yes, but you used NSInvocation in your fix. Moving to macro + @available as you did.